### PR TITLE
Feat/promisify

### PR DESCRIPTION
**Pull request by pvrooyen**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -149,134 +125,138 @@ RpcClient.prototype.batch = function (batchCallback, resultCallback) {
...
     return Array.prototype.slice.call(arr, start, end);
 };
 
-function generateRPCMethods(constructor, apiCalls, rpc) {
+function getRandomId() {
+    return parseInt(Math.random() * 100000);
+}
+
+const callSpec = {
+        abandonTransaction: 'str',//https://github.com/dashpay/dash/blob/v0.12.2.x/src/wallet/rpcwallet.cpp#L1896
```
Comment (antouhou) [inline lib/index.js]: Why there are _8_ spaces? :)

File: lib/promisify.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+function promisify(callbackFn) {
```
Comment (antouhou) [inline lib/promisify.js]: I thought that we agreed on using `bluebird`? It's in a task description. Bluebird BTW already has the promisifyAll method: http://bluebirdjs.com/docs/api/promise.promisifyall.html
So all changes that basically was needed is to create RPCClient the same way as it used to be, and return Bluebird.promisifyAll(rpcClient);

File: lib/index.js (hunk 1 above)
Comment (pvrooyen) [reply lib/index.js]: I think that was as per original.  Lint not enabled here yet which will cause a bigger refactor (out of scope)

File: lib/promisify.js (hunk 2 above)
Comment (pvrooyen) [reply lib/promisify.js]: As discussed in slack.  This to be redone then with bluebird.

**Comment by pvrooyen** [comment]
Makes more sense to create new PR than to revert this one. Do not merge this I will eventually delete.

**Comment by pvrooyen** [comment]
closed unmerged

**Pull request closed without merging**